extra/src/main/java/aima/extra/probability/domain/FiniteDomain.java

```diff
@@ -0,0 +1,45 @@
+package aima.extra.probability.domain;
+
+import java.util.Set;
+
+/**
+ * A Domain over a countable/discrete and finite set of objects.
+ * 
+ * @author Ciaran O'Reilly
+ * 
+ */
+public interface FiniteDomain extends DiscreteDomain {
+	/**
```

Comment:
should we add generics to the domain hierarchy for the types of their values? May be too restrictive.